### PR TITLE
fix(perf-issues) improve performance of the performance.issues_id query

### PR DESCRIPTION
**Pull request by DominikB2014**
Fixes PERF-1777 

Adds transaction name automatically to any performance.issue_ids query by looking it up via tagstorage.

### CODE REVIEW

File: src/sentry/search/events/datasets/discover.py [hunk 1]
```diff
@@ -1610,21 +1613,48 @@ def _performance_issue_ids_filter_converter(
...
+
+        transaction_limit = 4
+
+        groups: list[Group] = Group.objects.filter(id__in=value_list_as_ints)
+        transaction_tags: list[GroupTagValue] = tagstore.get_perf_group_list_tag_and_values(
+            groups, [], "transaction", transaction_limit, False
+        )
+        transaction_names: list[str] = list(map(lambda tag: tag.value, transaction_tags))
+
+        # perform_optimization = False
```
Comment (DominikB2014) [inline src/sentry/search/events/datasets/discover.py:1633]: Remove this comment

File: src/sentry/tagstore/snuba/backend.py [hunk 2]
```diff
@@ -460,6 +460,66 @@ def get_group_list_tag_value(self, project_ids, group_id_list, environment_ids,
...
+        group_ids = list(map(lambda group: group.id, groups))
+
+        filters = {"project_id": project_ids}
+
+        conditions = [[tag, "!=", ""]]
+        aggregations = [
+            ["uniq", tag, "values_seen"],
+            ["count()", "", "count"],
+            ["min", SEEN_COLUMN, "first_seen"],
+            ["max", SEEN_COLUMN, "last_seen"],
```
Comment (DominikB2014) [inline src/sentry/tagstore/snuba/backend.py:482]: Need to confirm what's required here, it was copied over for another place

**Review by evanh — COMMENTED** [review]
I do not recommend doing this.

I don't think this change is going to improve performance overall. The new `get_perf_group_list_tag_and_top_values` query is going to be very heavyweight, since it's querying all 90 days of data, filtering by the group_ids column which is not indexed and is slower for Clickhouse to process (Clickhouse has to unpack the array).

There is another query that does something similar to this (`tagstore.__get_tag_keys`). That query is by far the worst performing query in Snuba, and times out regularly. This version of that query will probably be slower because of the conditions on the `group_ids` column.

What was the idea behind this optimization? Is there not another way to find the transactions to filter by?

**Comment by DominikB2014** [comment]
> I do not recommend doing this.
> 
> I don't think this change is going to improve performance overall. The new `get_perf_group_list_tag_and_top_values` query is going to be very heavyweight, since it's querying all 90 days of data, filtering by the group_ids column which is not indexed and is slower for Clickhouse to process (Clickhouse has to unpack the array).
> 
> There is another query that does something similar to this (`tagstore.__get_tag_keys`). That query is by far the worst performing query in Snuba, and times out regularly. This version of that query will probably be slower because of the conditions on the `group_ids` column.
> 
> What was the idea behind this optimization? Is there not another way to find the transactions to filter by?

@evanh Thanks for the feedback, yes the idea was for optimization with the idea that finding the transaction tag and adding it to the discover query, is faster than just making the query. I'm not aware of any other more performant ways to get the transaction names other then by tags. 

But tags seems to be very fast, if you view the tags tab under a performance issue for example, i've never experienced a timeout.
Is your concern largely when there are multiple `performance.issue_ids`, having the optimization for a single issue id would suffice for now.

**Comment by DominikB2014** [comment]
Will Come back to this if further optimization is needed

**Pull request closed without merging**